### ISSUE TITLE
Support for mmm-based modes e.g. Vue mode

### ISSUE THREAD

**Issue by garyo** [joaotavora/eglot#475]
"Vue mode" is an mmm based (https://github.com/purcell/mmm-mode) major mode that switches between HTML, Typescript, and CSS modes based on which section of a single file you're editing. What this means with eglot is I get a different server depending on which section I'm in when I start eglot because `major-mode` is set per section.
Mmm does have a `mmm-primary-mode` which is set for the whole file (in my case it's set to "vue-mode"). 
Would it be possible to either check for `mmm-primary-mode` in `eglot--guess-contact` or maybe allow a function as the car of `eglot-server-programs` to return true if the contact in the cdr should be used in the current buffer? Or have an overridable function to get the major mode var to use?

**Comment by nemethf** [comment]
If  vls supports TCP connections, then (as a workaround) you start the server manually and later start eglot like C-u M-x eglot RET localhost:8080 RET. 

Alternatively, you can make a buffer-local or dir-local version of `eglot-server-programs` where html-mode, typescript-mode, etc are changed to start vls.

**Comment by garyo** [comment]
Sure, those are decent workarounds -- or Vue users can just always remember to start eglot with point in the HTML section of the file, typically at the top (where the major mode is indeed "vue-mode"). Just hoping for a more permanent solution.

**Comment by nemethf** [comment]
Does adding `(add-hook 'vue-mode-hook 'eglot-ensure)` to your init file work?  When you open a file the point is at the top, so the major mode should be vue-mode.

I understood you wanted a more robust solution, but you can continue to work with these until someone else provides that.

**Comment by garyo** [comment]
Yes, that does work, thanks! It only gets confused if I do `eglot-reconnect` or start it directly.

(By the way, if I do `eglot-reconnect` with point in a typescript section, I get a message in *Messages* saying it's still using the vue-mode server, `Server `EGLOT (horizon/vue-mode)' now managing `vue-mode' buffers in project `horizon'`, but it actually starts the typescript server. Weird corner case, no big deal.)

**Comment by joaotavora** [comment]
The solution for this is to use Rassumfrassum.  Have a look at this issue: https://github.com/joaotavora/rassumfrassum/issues/1  The `vuetail.py` bundled preset in Rassumfrassum uses Vue + Tailwind languages servers, and can be the basis of any other server combination you want to use.
